package/src/languageService/kustoLanguageService.ts

```diff
@@ -511,12 +557,101 @@ class KustoLanguageService implements LanguageService {
         });
     }
 
+    getClusterReferences(document: TextDocument, cursorOffset?: number): Promise<ClusterReference[]> {
```

Comment:
do you mind if we issue a teams channel poll to see what other developers think about the readability of the two?

Reply:
Followed up on your offline recommendation: 
```
if perf is most important, we can calculate this just once when changing the global state rather than every 200ms no?
``` 

Also added a way to test those methods.